packages/vuetify/src/components/VTreeview/VTreeview.ts

```diff
@@ -322,23 +332,31 @@ export default mixins(
 
       this.updateVnodeState(key)
     },
-    updateSelected (key: string | number, isSelected: boolean) {
+    updateSelected (key: string | number, isSelected: boolean, isForced = false) {
       if (!this.nodes.hasOwnProperty(key)) return
 
       const changed = new Map()
 
       if (this.selectionType !== 'independent') {
-        const descendants = [key, ...this.getDescendants(key)]
-        descendants.forEach(descendant => {
-          this.nodes[descendant].isSelected = isSelected
-          this.nodes[descendant].isIndeterminate = false
-          changed.set(descendant, isSelected)
+        this.getDescendants(key).forEach(descendant => {
+          if (!getObjectValueByPath(this.nodes[descendant].item, this.itemDisabled) || isForced) {
+            this.nodes[descendant].isSelected = isSelected
+            this.nodes[descendant].isIndeterminate = false
+            changed.set(descendant, isSelected)
+          }
         })
 
+        this.nodes[key].isSelected = isSelected
+        const calculated = this.calculateState(key, this.nodes)
```

Comment:
don't smush variables between logic. define as high as possible. same guidance as above.

Reply:
it has to be smushed because of logic